torch/csrc/jit/ir.cpp

```diff
@@ -1712,10 +1714,43 @@ at::ArrayRef<Value*> createTupleUnpack(Value* v) {
 
 std::vector<Value*> inlineCallTo(Node* to_replace, Function* callee) {
   WithInsertPoint guard(to_replace);
+  std::unordered_map<Value*, Value*> value_map;
   auto new_outputs = insertGraph(
       *to_replace->owningGraph(),
       *(callee->optimized_graph()),
-      to_replace->inputs());
+      to_replace->inputs(),
+      value_map);
+
+  std::unordered_map<InlinedCallStack*, InlinedCallStackPtr>
+      new_callstack_entries;
+
+  // TODO: We might need to use nodes_map instead of value_map. Otherwise, we
+  // are missing nodes without outputs (e.g. prim::Print).
+  std::unordered_set<Node*> updated_nodes;
+  for (const auto& kv : value_map) {
+    Node* new_node = kv.second->node();
+    if (!updated_nodes.insert(new_node).second) {
+      continue;
+    }
+
+    auto new_node_cs = new_node->callstack();
+
+    InlinedCallStack* raw_callstack_ptr =
```

Comment:
Isn't this just equivalent to `raw_callstack_ptr = new_node_cs->get()`?

Reply:
We have two dereferences there: first from Optional, second from intrusive_ptr to a raw pointer. The selection statement is for resolving the optional, I believe we cannot drop it.